File: public/dsa/arrays/pascals_triangle/iterative/pascals_triangle.py


from typing import List


class Solution:
    def pascalTriangleRow(self, rowNo: int) -> List[int]:
        result = []
        result.append(1)
        for i in range(2, rowNo + 1):
            prev = result[:]
            for j in range(1, len(prev)):
                result[j] = prev[j - 1] + prev[j]
            result.append(1)
        return result


# tests-start
rowNo = 5
sol = Solution()
print(sol.pascalTriangleRow(rowNo))
# tests-end

'''output
[1, 4, 6, 4, 1]
'''

File: public/dsa/arrays/pascals_triangle/iterative/pascals_triangle.cpp

#include <bits/stdc++.h>
using namespace std;

vector<int> pascalTriangleRow(int rowNo) {
    vector<int> result;
    result.push_back(1);
    for (int i = 2; i <= rowNo; i++) {
        auto prev = result;
        for (int j = 1; j < prev.size(); j++) {
            result[j] = prev[j - 1] + prev[j];
        }
        result.push_back(1);
    }
    return result;
}

// tests-start
int main() {
    int rowNo = 5;
    auto row = pascalTriangleRow(rowNo);
    for (int v : row) {
        cout << v << " ";
    }
    cout << endl;
    return 0;
}
// tests-end

/*output
1 4 6 4 1
*/